lightning/src/ln/channelmanager.rs

```diff
@@ -689,17 +696,28 @@ impl <Signer: ChannelSigner> PeerState<Signer> {
 	fn total_channel_count(&self) -> usize {
 		self.channel_by_id.len() +
 			self.outbound_v1_channel_by_id.len() +
-			self.inbound_v1_channel_by_id.len()
+			self.inbound_v1_channel_by_id.len() +
+			self.inbound_channel_request_by_id.len()
 	}
 
 	// Returns a bool indicating if the given `channel_id` matches a channel we have with this peer.
 	fn has_channel(&self, channel_id: &[u8; 32]) -> bool {
 		self.channel_by_id.contains_key(channel_id) ||
 			self.outbound_v1_channel_by_id.contains_key(channel_id) ||
-			self.inbound_v1_channel_by_id.contains_key(channel_id)
+			self.inbound_v1_channel_by_id.contains_key(channel_id) ||
+			self.inbound_channel_request_by_id.contains_key(channel_id)
 	}
 }
 
+/// A not-yet-accepted inbound (from counterparty) channel. Once
+/// accepted, the parameters will be used to construct a channel.
+pub(super) struct InboundChannelRequest {
+	/// The counterparty node id that initiated the channel open.
+	pub counterparty_node_id: PublicKey,
+	/// The original OpenChannel message.
+	pub open_channel_msg: msgs::OpenChannel,
+}
```

Comment:
So we'll need a new field here to track timer_ticks for expiry and then increment it on `timer_tick_occurred()` and expiring (removing the inbound requests with some logging) when they've reached some constant expiry ticks. Each tick is recommended to be ~1 minute so I believe even something as low as 2 is enough as we probably don't want to take so long to accept.